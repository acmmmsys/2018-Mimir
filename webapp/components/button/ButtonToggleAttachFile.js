import React from 'react';
import { connect } from 'react-redux';
import { compose, branch, renderComponent } from 'recompose';

import ButtonAction from 'components/button/ButtonAction';
import ButtonAttachFile from 'components/button/ButtonAttachFile';
import ButtonDetachFile from 'components/button/ButtonDetachFile';

const enhance = compose(
    connect(
        state => ({
            imageId: state.app.selectedImageId,
            image: state.app.images[state.app.selectedImageId],
        })
    ),
    branch(
        props => props.image && !props.image.attached,
        renderComponent(ButtonAttachFile),
        renderComponent(ButtonDetachFile)
    )
);

export default enhance(ButtonAction);