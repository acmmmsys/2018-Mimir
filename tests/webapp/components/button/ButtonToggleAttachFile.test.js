import React from 'react';
import ButtonToggleAttachFile from '../../../../webapp/components/button/ButtonToggleAttachFile'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('ButtonToggleAttachFile does render', () => {
  const wrapper = shallow(
      <ButtonToggleAttachFile store={mockStore(initialState)}>
        This is a test!
      </ButtonToggleAttachFile>
  );
  expect(wrapper).toMatchSnapshot();
});
