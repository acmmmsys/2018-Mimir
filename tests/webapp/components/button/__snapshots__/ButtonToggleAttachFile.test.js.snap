// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonToggleAttachFile does render 1`] = `
<branch(ButtonAction)
  dispatch={[Function]}
  imageId={null}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  This is a test!
</branch(ButtonAction)>
`;
